hyperspy/io.py

```diff
@@ -782,10 +782,40 @@ def save(filename, signal, overwrite=None, **kwds):
     elif overwrite is False and is_file:
         write = False  # Don't write the file
     else:
```

Comment:
When looking for missing coverage, I stumbled across the following segment. I do not really see, when the `else` statement would actually be called - except for when `overwrite` gets an invalid input, i.e. neither `bool` nor `None`. And then it partially repeats content from above.

From the history, this has been around fairly long @francisco-dlp , @ericpre

Reply:
This looks fine to me; it raises an error, when the input is not valid:
```python
import hyperspy.api as hs

s = hs.signals.Signal1D([0, 1, 2])
s.save('file.hspy', overwrite=0)
```
give the following error:
```python
Traceback (most recent call last):

  File "<ipython-input-1-f3574f48584c>", line 4, in <module>
    s.save('file.hspy', overwrite=0)

  File "/home/eric/Dev/hyperspy/hyperspy/signal.py", line 2840, in save
    io.save(filename, self, overwrite=overwrite, **kwds)

  File "/home/eric/Dev/hyperspy/hyperspy/io.py", line 776, in save
    raise ValueError(

ValueError: `overwrite` parameter can only be None, True or False.
```

Reply:
But in what case is all the code following the `else:` statement executed then?

Reply:
Ok, I see what you meant now, I was looking at the wrong `else` block... sorry! It looks like a the `RELEASE_next_minor` branch haven't been merged correctly after https://github.com/hyperspy/hyperspy/pull/2537 but I can't find the exact commit to be sure.
I agree with you, this doesn't make sense!

Reply:
Corrected in #2761